components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -489,7 +493,9 @@ private NotificationResponseBean resendAccountRecoveryNotification(User user, St
             secretKey = userRecoveryData.getSecret();
         } else {
             // Invalid previous confirmation code.
-            userRecoveryDataStore.invalidate(userRecoveryData.getSecret());
+            if (userRecoveryData != null) {
```

Comment:
Shall we use Stringutils isNotBlank()?

Reply:
Cannot use Stringutils.isNotBlank() as userRecoveryData is not a String.